dGame/dComponents/TriggerComponent.cpp

```diff
@@ -113,22 +160,169 @@ std::vector<Entity*> TriggerComponent::GatherTargets(LUTriggers::Command* comman
+}
+
+void TriggerComponent::HandlePushObject(Entity* targetEntity, std::vector<std::string> argArray){
+	auto* phantomPhysicsComponent = m_Parent->GetComponent<PhantomPhysicsComponent>();
+	if (!phantomPhysicsComponent) return;
+	phantomPhysicsComponent->SetPhysicsEffectActive(true);
+	phantomPhysicsComponent->SetEffectType(0); // push
+	phantomPhysicsComponent->SetDirectionalMultiplier(1);
+	NiPoint3 direction = NiPoint3::ZERO;
+	GeneralUtils::TryParse<float>(argArray.at(0), direction.x);
+	GeneralUtils::TryParse<float>(argArray.at(1), direction.y);
+	GeneralUtils::TryParse<float>(argArray.at(2), direction.z);
+	phantomPhysicsComponent->SetDirection(direction);
+
+	EntityManager::Instance()->SerializeEntity(m_Parent);
+}
+
+
+void TriggerComponent::HandleRepelObject(Entity* targetEntity, std::string args){
+	auto* phantomPhysicsComponent = m_Parent->GetComponent<PhantomPhysicsComponent>();
+	if (!phantomPhysicsComponent) return;
+	float forceMultiplier;
+	GeneralUtils::TryParse<float>(args, forceMultiplier);
+	phantomPhysicsComponent->SetPhysicsEffectActive(true);
+	phantomPhysicsComponent->SetEffectType(1); // repel/repulse
+	phantomPhysicsComponent->SetDirectionalMultiplier(forceMultiplier);
+
+	auto triggerPos = m_Parent->GetPosition();
+	auto targetPos = targetEntity->GetPosition();
+
+	// normalize the vectors to get the direction
+	auto delta = targetPos - triggerPos;
+	auto abs = sqrtf(delta.x * delta.x + delta.y * delta.y + delta.z * delta.z);
+	NiPoint3 direction = NiPoint3::ZERO;
+	direction.x = delta.x / abs;
+	direction.y = delta.y / abs;
+	direction.z = delta.z / abs;
+
+	phantomPhysicsComponent->SetDirection(direction);
+
+	EntityManager::Instance()->SerializeEntity(m_Parent);
+}
+
+void TriggerComponent::HandlePlayCinematic(Entity* targetEntity, std::vector<std::string> argArray) {
+	// [cinematic name],(lead-in in seconds),(“wait” to wait at end),(“unlock” to NOT lock the player controls),(“leavelocked” to leave player locked after cinematic finishes),(“hideplayer” to make player invisible during cinematic)
+	float leadIn = -1.0;
+	if (argArray.size() >= 2) GeneralUtils::TryParse<float>(argArray.at(1), leadIn);
+	int wait = 0;
+	if (argArray.size() >= 3 && argArray.at(2) == "wait") wait = 1;
+	bool unlock = true;
+	if (argArray.size() >= 4 && argArray.at(3) == "unlock") unlock = false;
+	bool leaveLocked = false;
+	if (argArray.size() >= 5 && argArray.at(4) == "leavelocked") leaveLocked = true;
+	bool hidePlayer = false;
+	if (argArray.size() >= 6 && argArray.at(5) == "hideplayer") hidePlayer = true;
+	GameMessages::SendPlayCinematic(targetEntity->GetObjectID(), GeneralUtils::UTF8ToUTF16(argArray.at(0)), targetEntity->GetSystemAddress(), true, true, false, false, 0, hidePlayer, leadIn, leaveLocked, unlock);
+}
+
+void TriggerComponent::HandleToggleBBB(Entity* targetEntity, std::string args) {
+	auto* character = targetEntity->GetCharacter();
+	if (!character) return;
+	bool buildMode = false;
```

Comment:
Should this default to false or the current build status?

Reply:
either, i have the logic for all cases, but could start with being !CurrentBuildStaus, since that's what I'm doing last